source/credentials_provider_sts_web_identity.c

```diff
@@ -0,0 +1,1120 @@
+            aws_error_str(aws_last_error()));
+        goto on_error;
+    }
+
+    config_profiles = aws_profile_collection_new_from_file(allocator, config_file_path, AWS_PST_CONFIG);
+    if (config_profiles != NULL) {
+        AWS_LOGF_DEBUG(
+            AWS_LS_AUTH_CREDENTIALS_PROVIDER,
+            "Successfully built config profile collection from file at (%s)",
+            aws_string_c_str(config_file_path));
+    } else {
+        AWS_LOGF_ERROR(
+            AWS_LS_AUTH_CREDENTIALS_PROVIDER,
+            "Failed to build config profile collection from file at (%s) : %s",
+            aws_string_c_str(config_file_path),
+            aws_error_str(aws_last_error()));
+        goto on_error;
+    }
+
+    aws_string_destroy(config_file_path);
+    return config_profiles;
+
+on_error:
+    if (config_file_path) {
+        aws_string_destroy(config_file_path);
+    }
+    if (config_profiles) {
+        aws_profile_collection_destroy(config_profiles);
+    }
+    return NULL;
+}
+
+static struct aws_byte_cursor s_default_profile_name_cursor = AWS_BYTE_CUR_INIT_FROM_STRING_LITERAL("default");
+static struct aws_byte_cursor s_default_region_cursor = AWS_BYTE_CUR_INIT_FROM_STRING_LITERAL("us-east-1");
+static struct aws_byte_cursor s_sts_cursor = AWS_BYTE_CUR_INIT_FROM_STRING_LITERAL("sts.");
+static struct aws_byte_cursor s_amazonaws_cursor = AWS_BYTE_CUR_INIT_FROM_STRING_LITERAL(".amazonaws.com");
+static struct aws_byte_cursor s_cn_cursor = AWS_BYTE_CUR_INIT_FROM_STRING_LITERAL(".cn");
+
+struct aws_credentials_provider_sts_web_identity_parameters *s_sts_web_identity_parameters_new(
+    struct aws_allocator *allocator) {
+
+    struct aws_credentials_provider_sts_web_identity_parameters *parameter =
+        aws_mem_calloc(allocator, 1, sizeof(struct aws_credentials_provider_sts_web_identity_parameters));
+    if (parameter == NULL) {
+        return NULL;
+    }
+    parameter->allocator = allocator;
+
+    bool success = false;
+    struct aws_string *region = NULL;
+    struct aws_string *role_arn = NULL;
+    struct aws_string *role_session_name = NULL;
+    struct aws_string *token_file_path = NULL;
+
+    /* check environment variables */
+    aws_get_environment_value(allocator, s_sts_web_identity_region_env, &region);
+    aws_get_environment_value(allocator, s_sts_web_identity_role_arn_env, &role_arn);
+    aws_get_environment_value(allocator, s_sts_web_identity_role_session_name_env, &role_session_name);
+    aws_get_environment_value(allocator, s_sts_web_identity_token_file_path_env, &token_file_path);
+
+    /**
+     * check config profile if either role_arn or token_file_path or role_session_name is not resolved from environment
+     * variable. Region is not enforced, but we need it to construct sts endpoint, if we can't find region from
+     * environment, we should check if it's set in config file, if neither, use default value "us-east-1" as region.
```

Comment:
It's worth re-evaluating/re-verifying if this kind of default region selection behavior is something we want to continue doing.  I've never liked it and we don't do it elsewhere in the CRT.